domain/history/usecase.go

```diff
@@ -1,35 +1,64 @@
 package history
 
 import (
+	"context"
+	"fmt"
 	"time"
 
-	"github.com/google/uuid"
-	"github.com/jackc/pgx/v4"
-
 	"github.com/alexyslozada/shorturl/model"
+
+	"github.com/google/uuid"
 )
 
 type History struct {
-	storage         Storage
-	useCaseShortURL UseCaseShortURL
+	storage  Storage
+	shortURL UseCaseShortURL
+	db       UseCaseDB
 }
 
 func New(s Storage, ucs UseCaseShortURL) History {
-	return History{storage: s, useCaseShortURL: ucs}
+	return History{storage: s, shortURL: ucs}
 }
 
-func (h History) CreateWithTx(tx pgx.Tx, m *model.History) error {
+func (h *History) SetUseCaseDB(useCase UseCaseDB) {
+	h.db = useCase
+}
+
+func (h History) hasUseCaseDB() bool {
+	return h.db != nil
+}
+
+func (h History) Create(m *model.History) error {
+	if !h.hasUseCaseDB() {
+		return fmt.Errorf("the db dependency is required")
+	}
+
 	m.ID = uuid.New()
 	m.CreatedAt = time.Now().Unix()
 
-	err := h.storage.CreateWithTx(tx, m)
+	tx, err := h.db.Tx()
 	if err != nil {
-		return err
+		return fmt.Errorf("h.db.Tx(): %v", err)
```

Comment:
Mejor cambiemos el `%v`

```suggestion
		return fmt.Errorf("h.db.Tx(): %w", err)
```